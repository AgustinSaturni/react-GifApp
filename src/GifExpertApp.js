import React, { useState } from 'react'
import { AddCategory } from './Componentes/AddCategory';
import { GifGrid } from './Componentes/GifGrid';

export const GifExpertApp = ({defaultCategorias=[]}) => {

//------Defino mi estado------------
    const [categorias, setCategorias] = useState(defaultCategorias); 
//-----------------------------------------------



//---------------Como agregar un elemento a un array-----------------------
    /* const AgregarElemento = () => {
        setCategorias([...categorias,'Dexter']); //Cuando uso el set, cambio el estado a lo que ponga dentro de los parentesis.
    } */
//-------------------------------

    return (
        <>
          <h2>GifExpertApp</h2>  
          <AddCategory setCategorias={setCategorias} />
          <hr/>
          
          <ol>
              {
                  categorias.map((categoria)=><GifGrid key={categoria} categoria={categoria} /> )
              }
          </ol>
        </>
    )
}
