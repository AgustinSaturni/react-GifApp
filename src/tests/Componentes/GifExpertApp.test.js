import { GifExpertApp } from "../../GifExpertApp";
import {shallow} from 'enzyme'
import React from 'react'

describe('Vamos a testear toda la App', ()=>{

    test('Match con el snapshot',()=>{
        const wrapper = shallow(<GifExpertApp />)  //el shallow renderiza un componente como el render.
        expect(wrapper).toMatchSnapshot(); 

    })

    test('Debe de mostrar una lista de categorias',()=>{

        const categorias=['One Punch','DragonBall'];
        const wrapper = shallow(<GifExpertApp defaultCategorias={categorias}/>)  //el shallow renderiza un componente como el render.
        expect(wrapper).toMatchSnapshot(); 
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    })

})