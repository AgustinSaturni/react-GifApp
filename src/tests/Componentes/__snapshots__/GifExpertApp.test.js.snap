// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vamos a testear toda la App Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="One Punch"
      key="One Punch"
    />
    <GifGrid
      categoria="DragonBall"
      key="DragonBall"
    />
  </ol>
</Fragment>
`;

exports[`Vamos a testear toda la App Match con el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
